src/Helpers/SwapperV2.sol

```diff
@@ -318,4 +271,73 @@ contract SwapperV2 is ILiFi {
 
         return balances;
     }
+
+    /// @dev Handles leftover tokens after swaps complete
+    /// @param _swaps Swap data array
+    /// @param _leftoverReceiver Address to send leftover tokens to
+    /// @param _initialBalances Array of initial token balances
+    /// @param _nativeReserve Amount of native token to prevent from being swept (0 for no reserve)
+    function _refundLeftovers(
+        LibSwap.SwapData[] calldata _swaps,
+        address payable _leftoverReceiver,
+        uint256[] memory _initialBalances,
+        uint256 _nativeReserve
+    ) private {
+        uint256 numSwaps = _swaps.length;
+        address finalAsset = _swaps[numSwaps - 1].receivingAssetId;
+
+        // Handle both intermediate receiving assets and leftover input tokens in a single loop
+        uint256 curBalance;
+        address curAsset;
+        address inputAsset;
+        uint256 curAssetReserve;
+        uint256 currentInputBalance;
+        uint256 inputAssetReserve;
+
+        for (uint256 i = 0; i < numSwaps; ) {
+            // Handle intermediate receiving assets (only for non-final swaps when numSwaps > 1)
+            if (i < numSwaps - 1 && numSwaps != 1) {
+                curAsset = _swaps[i].receivingAssetId;
+                // Handle multiple swap steps
+                if (curAsset != finalAsset) {
+                    curBalance =
```

Comment:
wouldnt sth like "swapOutput" be more accurate as variable name? It's not the current balance as far as I understand.